<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><link rel="canonical" href="https://ga4gh.github.io/wgs-qc/metrics_definitions/terminologies/" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>Terminologies & Concepts - GA4GH WGS Quality Control Standards</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Terminologies \u0026 Concepts";
        var mkdocs_page_input_path = "metrics_definitions/terminologies.md";
        var mkdocs_page_url = "/wgs-qc/metrics_definitions/terminologies/";
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> GA4GH WGS Quality Control Standards
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../overview/">Overview</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../motivation/">Motivation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../roadmap_v1/">Roadmap v1</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../roadmap_v2/">Roadmap v2</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../links/">Links</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Specifications</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../">About</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" >Post Alignment metrics</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../bases-q30/">Bases ≥ Q30</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../cross-contamination/">Cross contamination</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../genome-coverage-uniformity/">Genome coverage uniformity</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../insert-size-standard-deviation/">Insert size standard deviation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../mean-autosome-coverage/">Mean autosome coverage</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../mean-insert-size/">Mean insert size</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../percent-autosomes-covered-15-x/">Percent autosomes covered ≥ 15 X</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../percent-reads-mapped/">Percent reads mapped</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../percent-reads-properly-paired/">Percent reads properly paired</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" >Post Variant calling metrics</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../count-deletions/">Count: Deletions</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../count-insertions/">Count: Insertions</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../count-snvs/">Count: SNVs</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../ratio-heterozygous-homozygous-indels/">Ratio: Heterozygous/Homozygous (indels)</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../ratio-heterozygous-homozygous-snvs/">Ratio: Heterozygous/Homozygous (SNVs)</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../ratio-insertions-deletions/">Ratio: Insertions/Deletions</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../ratio-transitions-transversions-ti-tv/">Ratio: Transitions/Transversions (ti/tv)</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="#">Terminologies & Concepts</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#high-quality-reads">High quality reads</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#primary-alignments">Primary alignments</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#base-quality-score">Base quality score</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#mapping-quality">Mapping quality</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#phred-scale">Phred scale</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#duplicated-reads">Duplicated reads</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#clipped-bases">Clipped bases</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#picard-collectqualityyieldmetrics">Picard CollectQualityYieldMetrics</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#picard-collectwgsmetrics">Picard CollectWgsMetrics</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#autosomes-non-gap-regions">Autosomes non gap regions</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#grch38-assembly">GRCh38 assembly</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#overlapping-bases">Overlapping bases</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#high-quality-variants">High quality variants</a>
    </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../References/">References</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Implementations</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../implementations/">About</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../implementations/npm-sample-qc_implementations/">NPM-sample-qc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../implementations/argo-dnalanqc_implementations/">ICGC-ARGO-dnaalnqc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../implementations/argo-vcfqc_implementations/">ICGC-ARGO-vcfqc</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Implementation Benchmarking</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../benchmark_resources/">About</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../benchmark_resources/benchmark_resources_data/">Resources</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../benchmark_resources/benchmark_resources_notebook/">Jupyter Notebook</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">GA4GH WGS Quality Control Standards</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Specifications</li>
      <li class="breadcrumb-item active">Terminologies & Concepts</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/ga4gh/quality-control-wgs/edit/master/docs/metrics_definitions/terminologies.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h2 id="terminologies-concepts">Terminologies &amp; Concepts</h2>
<h3 id="high-quality-reads">High quality reads</h3>
<p>High quality reads refers to sequencing instrument filter. For Illumina sequencer, leverages internal quality filtering procedure called <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890991-PF-reads-Illumina-chastity-filter">chastity filter</a>. Reads that pass this filter are called "PF" for "pass-filter"</p>
<h3 id="primary-alignments">Primary alignments</h3>
<p>Primary alignments refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.2 ”Terminologies and Concepts”, “Multiple mapping“ &amp; section 1.4 “The alignment section: mandatory fields”, “FLAG: Combination of bitwise FLAGs”</p>
<h3 id="base-quality-score">Base quality score</h3>
<p>Base quality score refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.4: “The alignment section: mandatory fields”, “QUAL”</p>
<h3 id="mapping-quality">Mapping quality</h3>
<p>Mapping qualityscore refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.4 “The alignment section: mandatory fields”, “MAPQ: MAPping Quality”</p>
<h3 id="phred-scale">Phred scale</h3>
<p>PHRED scale refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.2 ”Terminologies and Concepts”, “Phred scale“</p>
<h3 id="duplicated-reads">Duplicated reads</h3>
<p>Duplicated reads refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.4: “The alignment section: mandatory fields”, “FLAG: Combination of bitwise FLAGs”</p>
<h3 id="clipped-bases">Clipped bases</h3>
<p>Clipped bases refers to <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM Format Specification v1</a>, section 1.2: ”Terminologies and Concepts”, “Linear alignment“ &amp; section 1.4 “The alignment section: mandatory fields”, “CIGAR: CIGAR string”</p>
<h3 id="picard-collectqualityyieldmetrics">Picard CollectQualityYieldMetrics</h3>
<p>Picard CollectQualityYieldMetrics refers to <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360040507031-CollectQualityYieldMetrics-Picard-">GATK documentation</a>, section "CollectQualityYieldMetrics (Picard)" &amp; <a href="https://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectQualityYieldMetrics.QualityYieldMetrics">GATK specifications</a></p>
<h3 id="picard-collectwgsmetrics">Picard CollectWgsMetrics</h3>
<p>Picard CollectWgsMetrics refers to <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036804671-CollectWgsMetrics-Picard">GATK documentation</a>, section "CollectWgsMetrics (Picard)" &amp; <a href="https://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectWgsMetrics.WgsMetrics">GATK specifications</a></p>
<h3 id="autosomes-non-gap-regions">Autosomes non gap regions</h3>
<p>Autosomes non gap regions refers to the selection of chromosome 1 to 22, excluding chromosome X, Y, M and alternative contigs. In addition, gap regions as defined by UCSC are excluded. See <a href="https://github.com/c-BIG/NPM-sample-qc/blob/master/resources/README.rst">NPM-sample-qc documentation</a></p>
<h3 id="grch38-assembly">GRCh38 assembly</h3>
<p>GRCh38 assembly refers to <a href="https://1000genomes-dragen-3.7.6.s3.amazonaws.com/references/fasta/hg38.fa">1000genome-dragen-3.7.6 reference</a></p>
<h3 id="overlapping-bases">Overlapping bases</h3>
<p>Overlapping bases refers to <a href="https://github.com/brentp/mosdepth">mosdepth documentation</a>, section "how it works"</p>
<h3 id="high-quality-variants">High quality variants</h3>
<p>High quality variants (<code>PASS FILTER</code>) refers to <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF Format Specicification v4.2</a>, section 1.4.1 "FILTER - filter status:" &amp; <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants">GATK generic hard-filtering recommendations</a></p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../ratio-transitions-transversions-ti-tv/" class="btn btn-neutral float-left" title="Ratio: Transitions/Transversions (ti/tv)"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../References/" class="btn btn-neutral float-right" title="References">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
      <p>GA4GH Large Scale Genomics Workstream, WGS-QC Standards team</p>
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/ga4gh/quality-control-wgs" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../ratio-transitions-transversions-ti-tv/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../References/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
